### PR TITLE
Provide self contained npm scripts

### LINKED ISSUE #154: Provide self contained npm scripts
## Enhancement 
Dojo CLI applications should be easy to run. npm scripts are easy for people to use who may not be familiar with the Dojo CLI and can run dev dependencies that don't need to be installed separately (i.e. globally)

### Expected behavior:
An application generated using cli-create-app should be able to be installed an run easily when git cloned. Ideally, some npm scripts should be runnable immediately after install.

### Actual behavior:
A cloned repo generated from the cil-create-app requires the domo cli to be installed to run.

### Current template
https://github.com/dojo/cli-create-app/blob/fbd9118d30ee7a7d6483d6ed5814e231e331d441/src/templates/package.json#L1-L14

### Suggested Template
```json
{
  "name": "<%- appName %>",
  "version": "1.0.0",
  "scripts": {
    "dev": "dojo build --mode dev --watch memory --serve",
    "build": "dojo build",
    "test": "dojo build -m test && dojo test -a"
  },
  "dependencies": {
    "@dojo/framework": "~3.0.0",
    "tslib": "~1.8.1"
  },
  "devDependencies": {
    "@dojo/cli": "^3.0.0",
    "@dojo/cli-build-app": "~3.0.0",
    "@dojo/cli-test-intern": "~3.0.0",
    "@types/node": "~9.6.5",
    "typescript": "~2.6.1"
  }
}
```

### PR DESCRIPTION
**Pull request by KaneFreeman**
**Type:** feature

The following has been addressed in the PR:

* [x] There is a related issue
* [x] All code has been formatted with [`prettier`](https://prettier.io/) as per the [readme code style guidelines](./../#code-style)
* [ ] Unit or Functional tests are included in the PR

**Description:**
Adds in dev, build and test scripts, along with adding @dojo/cli as a dev dependency, to the package.json templates.

Resolves #154

### CODE REVIEW

**Comment by aciccarello** [comment]
I would recommend not using the abbreviated flags to make the scripts more intuitive for new users.

**Comment by KaneFreeman** [comment]
@aciccarello the scripts have been updated with their long formats.

**Comment by KaneFreeman** [comment]
@aciccarello I have updated the scripts to their long form.

**Comment by agubler** [comment]
@KaneFreeman I guess we should add something about the scripts to the template README files?

**Comment by KaneFreeman** [comment]
I have updated the README files to include mention of the new scripts, while still mentioning the `@dojo/cli` equivalent. I added a `build:dev` script in as well since it is explicitly called out in the READMEs.

File: src/templates/ts/skeleton/README.md [hunk 1]
```diff
@@ -4,29 +4,37 @@ This project was generated with the [Dojo CLI](https://github.com/dojo/cli) & [D
...
+To create an in memory development build and start a development server with hot reload, switch the `--watch` option to `memory`.
 
 ## Running unit tests
 
 To run units tests in node only use `dojo test` which uses JIT (just in time) compilation.
 
-To run the unit tests against built bundles, first the run a test build with `dojo build --mode unit`. The build test artifacts are written to the `output/tests/unit` directory.
+To run the unit tests against built bundles, run `npm run test:unit`.
+
+To be re-run the unit tests without needing to re-build the full application each time, first build the app with `--most unit` and the `--watch` option, `dojo build --mode unit --watch`. Then run `dojo test --config local` to run the unit tests as needed.
```
Comment (agubler) [inline src/templates/ts/skeleton/README.md]: Should this be `--mode unit`?

Comment (KaneFreeman) [reply src/templates/ts/skeleton/README.md]: Lol, yeah. I have updated that.

File: src/run.ts [hunk 2]
```diff
@@ -17,6 +17,15 @@ export default async function(helper: Helper, args: CreateAppArgs) {
 	const isSkeleton = args.skeleton;
 	const isTsx = args.tsx;
 
+	const scripts = {
+		dev: 'dojo build --mode dev --watch file --serve',
+		build: 'dojo build --mode dist',
+		buildDev: `dojo build --mode dev`,
+		test: 'npm run test:unit && npm run test:functional',
```
Comment (agubler) [inline src/run.ts]: I think that `npm run test` should translate to `dojo test` that doesn't need a bundle to be built as it does transpilation on the fly.

Perhaps the others:
 * `npm run test:unit` should be `dojo build --mode unit && dojo test --unit --config local` 
 * `npm run test:functional` should be `dojo build --mode functional && dojo test --functional --config local` 
 * `npm run test:all` should be `dojo build --mode unit && dojo build --mode functional && dojo test --all --config local`

Comment (KaneFreeman) [reply src/run.ts]: @agubler The proposed changes have been made.

**Review by agubler — APPROVED** [review]

**Pull request merged**